packages/jss/src/renderers/DomRenderer.js

```diff
@@ -216,45 +216,66 @@ function findCommentNode(text: string): Node | null {
   return null
 }
 
+type PrevNode = {
+  parent: ?Node,
+  node: ?Node
+}
+
 /**
  * Find a node before which we can insert the sheet.
  */
-function findPrevNode(options: PriorityOptions): ?Node | null {
+function findPrevNode(options: PriorityOptions): PrevNode | false {
   const {registry} = sheets
 
   if (registry.length > 0) {
     // Try to insert before the next higher sheet.
     let sheet = findHigherSheet(registry, options)
-    if (sheet) return sheet.renderer.element
+    if (sheet) {
+      return {
```

Comment:
Seems like calling parentNode in advance doesn't bring much value other than makes findPrevNode semantics wrong. Mb I oversee  something but to me it looks like we could keep the return value as it was.

Reply:
The problem is when returning nextSibling and when it returns `null` we can't get the parentNode

Reply:
Not necessarily, it was supposed to work with appending it at the end of the parent of the stylesheet